io/src/read_cells.cpp

```diff
@@ -19,6 +19,7 @@ namespace traccc::io {
 
 void read_cells(edm::silicon_cell_collection::host& cells, std::size_t event,
                 std::string_view directory,
+                std::unique_ptr<const Logger> ilogger,
```

Comment:
For standalone functions, would it not be acceptable to just use a hardcoded logger name? :thinking: For classes of course we want to have the ability to provide the logging name from the outside. But for stateless functions, is it necessary? Would it not be okay to just use `"traccc::io::read_cells"`, `"traccc::io::csv::read_cells"`, etc. for the source of the messages? :thinking:

Then we wouldn't need to bother with changing the API of so many of these functions.

Reply:
This is not quite how loggers work here. The loggers carry much more than just a name, they also carry formatting and configuration options. I agree we can accept a default argument, but the only valid default argument is the dummy logger; we cannot have a default name for loggers as you suggest.

Reply:
That's unfortunate. Is it really not possible to have a default logger with some specific name, that prints messages to `std::cout` with level `INFO`? :confused: I thought the idea was that if one was okay with messages going "with the Acts formatting" to `std::cout`, they wouldn't need to bother too much with setting up their own loggers. Only when they want to pipe messages "into their own system", would they need to become explicit.

I accept that even the I/O functions need to be able to receive a logger from the outside. (Though I'll try to think of a better API for that in a follow-up.) But why can't the default still be something like

```
= getDefaultLogger("meaningfulDefaultName", Logging::INFO)
```

? Why must it be a dummy logger by default?

Reply:
It's possible, but we should not risk writing anything to `std::cout` without explicit user instruction to do so. That's why the only reasonable default logger is the dummy one.

Reply:
Why? :confused: As long as there is a clear and well documented way for how users can override it, what's the harm in having default messages be meaningful/pretty? At that point, why use `getDummyLogger()` at all? Why don't we just force users to **always** supply a logger that they constructed?

Reply:
> Why don't we just force users to **always** supply a logger that they constructed?

That's exactly the current state of the code. :D

Reply:
Default added here.